onnxruntime/core/framework/op_kernel.cc

```diff
@@ -196,12 +196,14 @@ Status OpKernelContext::SetOutputMLValue(int index, const OrtValue& ort_value) {
   if (index < 0 || index >= OutputCount()) {
     return Status(common::ONNXRUNTIME, common::FAIL,
                   "Index out of range. " + std::to_string(index) +
-                  " was specified, but " + "range is (0, " + std::to_string(OutputCount()) + ")");
+                      " was specified, but " + "range is (0, " + std::to_string(OutputCount()) + ")");
   }
 
-  ORT_ENFORCE(kernel_->KernelDef().HasExternalOutputs(),
-              GetOpType() + " is trying to use SetOutputMLValue(), but its kernel_def doesn't have "
-              ".ExternalOutputs() declared.");
+  if (kernel_->KernelDef().HasExternalOutputs()) {
+    std::cout << "Node name: " << GetNodeName() << ", node type: " << GetOpType()
+              << " is trying to use SetOutputMLValue(), but its kernel_def doesn't have "
+              << ".ExternalOutputs() declared." << std::endl;
+  }
 
```

Comment:
AtenOp PR removed this. 
I think it's probably still better to keep this. 

Please use logger_default?